src/content/docs/en/reference/dev-toolbar-app-reference.mdx

```diff
@@ -9,271 +9,327 @@ The Astro Dev Toolbar App API allows you to create apps that can be used to exte
 
 ```ts title="src/my-app.js" {3-5}
-export default {
-	// ...
+export default defineToolbarApp({
 	init(canvas) {
     canvas.appendChild(document.createTextNode('Hello World!'))
   }
-}
+});
 ```
 
-#### `eventTarget`
+#### `app`
+
+An [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) augmented with a few helpers that can be used to send and receive events from the Dev Toolbar, see the [Client-side Events](#client-side-events) section for more information.
+
+```ts title="src/my-app.js" {3-8}
+export default defineToolbarApp({
+  init(canvas, app) {
+    app.onAppToggled(({ state }) => {
+      const text = document.createTextNode(
+        `The app is now ${state ? "enabled" : "disabled"}!`,
+      );
+      canvas.appendChild(text);
+    });
+  },
+});
+```
 
-An [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) that can be used to send and receive events from the Dev Toolbar.
+#### `server`
 
-### `beforeTogglingOff`
+An object that can be used to communicate with the server, see the [Client-Server Communication](#client-server-communication) section for more information.
 
-This optional function will be called when the user clicks on the app icon in the UI to toggle off the app. This function can be used, for example, to perform cleanup operations, do an animation, or to ask the user for confirmation before toggling off the app.
+```ts title="src/my-app.js" {3-7}
+export default defineToolbarApp({
+  init(canvas, app, server) {
+    server.send('my-message', { message: 'Hello!' });
+
+    server.on('server-message', (data) => {
+      console.log(data.message);
+    });
+  },
+});
+```
+
+### `beforeTogglingOff()`
+
+**Signature:** `beforeTogglingOff(canvas: ShadowRoot): boolean | void`
+
+This optional function will be called when the user clicks on the app icon in the UI to toggle off the app. This function can be used, for example, to perform cleanup operations, or to ask the user for confirmation before toggling off the app.
 
 If a falsy value is returned, the toggling off will be cancelled and the app will stay enabled.
 
 ```ts title="src/my-app.js" {3-6}
-export default {
+export default defineToolbarApp({
   // ...
   beforeTogglingOff() {
     const confirmation = window.confirm('Are you sure you want to disable this app?');
     return confirmation;
   }
-}
+});
 ```
 
 #### canvas
 
-The ShadowRoot of the app, can be used to render any UI needed.
+The ShadowRoot of the app, can be used to render any UI needed before closing. Same as the `canvas` argument of the `init` function.
 
 ## Client-side Events
 
-Using the [`eventTarget`](#eventtarget) argument on the [`init`](#init-canvas-shadowroot-eventtarget-eventtarget--void) hook, apps can send and receive events from the Dev Toolbar. The following events are available:
+In addition to the standard methods of an `EventTarget` (addEventListener, dispatchEvent, etc), the `app` object also has the following methods:
```

Comment:
Note here that I think we've lost the previous example showing `eventTarget.addEventListener` which is probably a helpful one to show! I don't mind if it's also here!

also, just double checking capital E on `EventTarget` here in this context!

Reply:
It's okay to omit it, there's no point (at this time) to use the normal `addEventListener` method, the helpers methods cover the entire API.

`EventTarget` is also correct, as it refer to the class here.